examples/hosts/app-integration/external-data/src/mock-customer-service/service.ts

```diff
@@ -0,0 +1,189 @@
+import fetch from "node-fetch";
+import { isWebUri } from "valid-url";
+
+import { MockWebhook } from "../utilities";
+import { assertValidTaskData, TaskData } from "../model-interface";
+
+/**
+ * {@link initializeCustomerService} input properties.
+ */
+export interface ServiceProps {
+	/**
+	 * Port to listen on.
+	 */
+	port: number | string;
+
+	/**
+	 * URL of the external data service webhook.
+	 *
+	 * @remarks
+	 *
+	 * This service will register with the external data service to receive data change notifications.
+	 * Any restered listeners for this "webhook echo" via `/register-for-webhook` will be notified of changes
+	 * any time the external data service communicates them.
+	 */
+	externalDataServiceWebhookRegistrationUrl: string;
+}
+
+/**
+ * Initializes the mock customer service.
+ *
+ * @remarks Consumers are required to manually dispose of the returned `Server` object.
+ */
+export async function initializeCustomerService(props: ServiceProps): Promise<Server> {
+	const { port, externalDataServiceWebhookRegistrationUrl } = props;
+
+	/**
+	 * Helper function to prepend service-specific metadata to messages logged by this service.
+	 */
+	function formatLogMessage(message: string): string {
+		return `CUSTOMER SERVICE (${port}): ${message}`;
+	}
+
+	/**
+	 * Mock webhook for echoing webhook notifications from the external data service.
+	 */
+	const webhook = new MockWebhook<TaskData>();
+
+	// Register with external data service for webhook notifications.
+	try {
+		await fetch(externalDataServiceWebhookRegistrationUrl, {
+			method: "POST",
+			headers: {
+				"Access-Control-Allow-Origin": "*",
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify({
+				// External data service will call our webhook echoer to notify our subscribers of the data changes.
+				url: `http://localhost:${port}/echo-external-data-webhook`,
+			}),
+		});
+	} catch (error) {
+		console.error(
+			formatLogMessage(
+				`Registering for data update notifications webhook with the external data service failed due to an error.`,
+			),
+			error,
+		);
+		throw error;
+	}
+
+	const expressApp = express();
+	expressApp.use(express.json());
+	expressApp.use(cors());
+
+	expressApp.get("/", (_, result) => {
+		result.send();
+	});
+
+	/**
+	 * Register's the sender's URL to receive notifications when the external task-list data changes.
+	 *
+	 * Expected input data format:
+	 *
+	 * ```json
+	 * {
+	 *  url: string // The target URL to receive change notification
+	 * }
+	 * ```
+	 *
+	 * Notifications sent to subscribers will contain the updated task-list data in the form of:
+	 *
+	 * ```json
+	 * {
+	 *  taskList: {
+	 *      [id: string]: {
+	 *          name: string,
+	 *          priority: number
+	 *      }
+	 *  }
+	 * }
+	 * ```
+	 */
+	expressApp.post("/register-for-webhook", (request, result) => {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		const subscriberUrl = request.body.url as string;
```

Comment:
```suggestion
		const subscriberUrl = request?.body?.url as string;
```
And same question about express types from the other file.